File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/03_recursion/julia/02_greet.jl
using Test
using Suppressor

function greet2(name)
    println("how are you, ",name,"?")
end

function bye()
    println("ok bye!")
end

function greet(name)
    println("hello, ",name,"!")
    greet2(name)
    println("getting ready to say bye...")
    bye()
end

result = @capture_out greet("adit")
@test result == "hello, adit!
how are you, adit?
getting ready to say bye...
ok bye!
"

File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/03_recursion/c++11/02_greet.cpp
#include <iostream>
#include <string>

using std::cout;
using std::endl;

void greet2(std::string name) {
    cout << "How are you, " + name + "?" << endl;
}

void bye() {
    cout << "Ok, bye!" << endl;
}


void greet(std::string name) {
    cout << "Hello, " + name + "!" << endl;
    greet2(name);
    cout << "Getting ready to say bye..." << endl;
    bye();
}


int main() {
    greet("Adit");

    return 0;
}